From 7fab374df7ce28512caee5f1890c82c0f0bf07d3 Mon Sep 17 00:00:00 2001
From: rleix <rayx.lei@intel.com>
Date: Wed, 27 Jun 2018 12:50:04 +0800
Subject: [PATCH] Fix the sync issue between HVAC app and KitchSink app

Currently cel_apl used the default vehiclehal from Google, there's no
real vehicle hardware/system and it just used to simulate the vehiclehal,
when we set a property to vehicle, there's no feedback.
So when set a value on HVAC app or KitchSink app, there will be no notification
to the other app. It caused the value not match between these two applications.

Change-Id: I7deefbf7d641cd09594ae8a8e44fab77d426ac95
Jira: OAM-65770
Test: Refer to the steps from JIRA
Signed-off-by: Lei,RayX <rayx.lei@intel.com>
---
 .../vehicle/2.0/default/impl/vhal_v2_0/EmulatedVehicleHal.cpp    | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/automotive/vehicle/2.0/default/impl/vhal_v2_0/EmulatedVehicleHal.cpp b/automotive/vehicle/2.0/default/impl/vhal_v2_0/EmulatedVehicleHal.cpp
index 07695bf..d66ac45 100644
--- a/automotive/vehicle/2.0/default/impl/vhal_v2_0/EmulatedVehicleHal.cpp
+++ b/automotive/vehicle/2.0/default/impl/vhal_v2_0/EmulatedVehicleHal.cpp
@@ -186,6 +186,15 @@ StatusCode EmulatedVehicleHal::set(const VehiclePropValue& propValue) {
 
     if (!mPropStore->writeValue(propValue, shouldUpdateStatus)) {
         return StatusCode::INVALID_ARG;
+    } else {
+       /*
+        * w/a for the sync issue of HVAC app and KitchSink app.
+        * Currently, there's no real vehicle system for our platform, when we set a property
+        * to vehicle, there's no feedback. So when set a value on HVAC app or KitchSink app,
+        * there will no notification to the other app.
+        * We report a HAL event here to w/a this issue.
+        */
+        doHalEvent(getValuePool()->obtain(propValue));
     }
 
     getEmulatorOrDie()->doSetValueFromClient(propValue);
-- 
1.9.1

